dc-commons-xml/src/main/java/de/digitalcollections/commons/xml/xpath/DocumentReader.java

```diff
@@ -300,6 +300,40 @@ private String resolveVariableContext(
     }
   }
 
+  // TODO Don't restrict to string lists
+  private List<String> resolveVariablesAsStrings(String[] paths, boolean multiValued) {
+    List<String> result = new ArrayList<>();
+    paths = prependWithRootPaths(paths);
+
+    for (String path : paths) {
+      for (Object resolvedObject : xpw.asListOfObjects(path)) {
+
+        if (resolvedObject instanceof String && !((String) resolvedObject).isEmpty()) {
+          result.add((String) resolvedObject);
+          continue;
+        }
+        if (resolvedObject instanceof Integer) {
+          continue;
```

Comment:
Wäre es nicht besser hier nen String draus zu basteln?

```suggestion
          result.add(((Integer) resolvedObject).toString());
          continue;
```

Reply:
Danke, die Zeile ist irgendwie beim Commit herausgefallen :)